llvm/lib/Transforms/Vectorize/VPlan.cpp

```diff
@@ -1007,9 +1007,19 @@ void VPlan::execute(VPTransformState *State) {
          "middle block has unexpected successors");
   VPBasicBlock *ScalarPhVPBB = cast<VPBasicBlock>(
       MiddleSuccs.size() == 1 ? MiddleSuccs[0] : MiddleSuccs[1]);
-  assert(!isa<VPIRBasicBlock>(ScalarPhVPBB) &&
-         "scalar preheader cannot be wrapped already");
-  replaceVPBBWithIRVPBB(ScalarPhVPBB, ScalarPh);
+  if (!isa<VPIRBasicBlock>(ScalarPhVPBB)) {
+    assert(!isa<VPIRBasicBlock>(ScalarPhVPBB) &&
```

Comment:
Isn't this testing exactly the same thing in the if statement? I think you can remove this as being redundant.

Reply:
Yep, removed the assert, thanks!